lib/matplotlib/tests/test_patches.py

```diff
@@ -54,6 +54,41 @@ def test_Polygon_close():
     assert_array_equal(p.get_xy(), xyclosed)
 
 
+def test_corners_edges():
+    loc = [10, 20]
+    width = 1
+    height = 2
+
+    corners = ((10, 20), (11, 20), (11, 22), (10, 22))
+    edges = ((10, 21), (10.5, 22), (11, 21), (10.5, 20))
+    rect = Rectangle(loc, width, height)
+    assert_array_equal(rect.get_corners(), corners)
+    assert_array_equal(rect.get_edge_midpoints(), edges)
+    assert_array_equal(rect.get_center(), (10.5, 21))
+
+    corners_rot = ((10, 20), (10, 21), (8, 21), (8, 20))
+    edges_rot = ((9, 20), (8, 20.5), (9, 21), (10, 20.5))
+    rect.set_angle(90)
```

Comment:
Wondering if this is worth using an example of rotation less symmetrical? It may be slightly more annoying to get the expected values but at least it has the benefit of catching bugs more easily.

Reply:
👍 will add one it

Reply:
Added